dpnp/dpnp_algo/dpnp_elementwise_common.py

```diff
@@ -43,9 +43,432 @@
     "DPNPReal",
     "DPNPRound",
     "DPNPUnaryFunc",
+    "ufunc",
 ]
 
 
+import dpctl.tensor._tensor_elementwise_impl as ti
+
+import dpnp.backend.extensions.vm._vm_impl as vmi
+
+
+class ufunc:
+    """
+    ufunc(name, docs, nin, ...)
+
+    DPNP provides universal functions (a.k.a. ufuncs) to support various
+    element-wise operations.
+
+    Optional keyword arguments
+    --------------------------
+    out : {None, dpnp.ndarray, usm_ndarray}, optional
+        Output array to populate.
+        Array must have the correct shape and the expected data type.
+    order : {None, "C", "F", "A", "K"}, optional
+        Memory layout of the newly output array, Cannot be provided
+        together with `out`. Default: ``"K"``.
+    dtype : {None, dtype}, optional
+        If provided, the destination array will have this dtype. Cannot be
+        provided together with `out`. Default: ``None``.
+    casting : {"no", "equiv", "safe", "same_kind", "unsafe"}, optional
+        Controls what kind of data casting may occur. Cannot be provided
+        together with `out`. Default: ``"safe"``.
+
+    Limitations
+    -----------
+    Keyword arguments `where` and `subok` are supported with their default values.
+    Other keyword arguments is currently unsupported.
+    Otherwise ``NotImplementedError`` exception will be raised.
+
+    """
+
+    def __init__(
+        self,
+        name,
+        docs,
+        nin,
+        mkl_call=False,
+        inplace=False,
+        acceptance_fn=False,
+    ):
+        self.name_ = name
+        self.nin_ = nin
+        _name = "_" + name
+
+        dpt_result_type = getattr(ti, _name + "_result_type")
```

Comment:
How will it be propose to handle elementwise functions which are not present in dcptl? (like `fabs` which we will need to implement separately in dpnp)